src/MDOP.jl

```diff
@@ -87,3 +220,73 @@ function beta_statistic(Y::Dataset, s::Vector; τs = 0:50 , w::Int = 1)
     β = mean(log10.(Δϕ); dims=1)
     return vec(β)
 end
+
+
+"""
+    embed_one_cycle(Y, s::Vector, τ::Int) -> Z
+Add the `τ` lagged values of the time series `s` as additional component to `Y`
+(`Vector` or `Dataset`), in order to form a higher embedded
+dataset `Z`. The dimensionality of `Z` is thus equal to that of `Y` + 1.
+"""
+function embed_one_cycle(Y::Dataset{D,T}, s::Vector{T}, τ::Int) where {D, T<:Real}
+    N = length(Y)
+    @assert N ≤ length(s)
+    M = N - τ
+    data = Vector{SVector{D+1, T}}(undef, M)
+    @inbounds for i in 1:M
+        data[i] = SVector{D+1, T}(Y[i]..., s[i+τ])
+    end
+    return Dataset(data)
+end
+
+function embed_one_cycle(Y::Vector{T}, s::Vector{T}, τ::Int) where {T<:Real}
+    N = length(Y)
+    @assert N ≤ length(s)
+    M = N - τ
+    return Dataset(view(s, 1:M), view(s, τ+1:N))
+end
+
+
+
+"""
+    estimate_maximum_delay(s::Vector; tw=1:50, samplesize::Float=1.0)) -> τ_max, L
+Compute an upper bound for the search of optimal delays, when using `MDOP`
+[`MDOP`](@ref) or `beta_statistic` [`beta_statistic`](@ref).
+
+## Description
+The input time series `s` gets embedded with unit lag and increasing dimension,
+for dimensions (or time windows) `tw` (`RangeObject`).
+For each of such a time window the `L`-statistic from Uzal et al. [^Uzal2011]
+will be computed. `samplesize` determines the fraction of points to be
+considered in the computation of `L` (see [`uzal_cost`](@ref)). When this
+statistic reaches its global minimum the maximum delay value `τ_max` gets
+returned. When `s` is a multivariate `Dataset`, `τ_max` will becomputed for all
+time series of that Dataset and the maximum value will be returned. The returned
+`L`-statistic is a Dataset of size (length(tw)*size(s,2)).
+
+[^Nichkawde2013]: Nichkawde, Chetan (2013). [Optimal state-space reconstruction using derivatives on projected manifold. Physical Review E 87, 022905](https://doi.org/10.1103/PhysRevE.87.022905).
+[^Uzal2011]: Uzal, L. C., Grinblat, G. L., Verdes, P. F. (2011). [Optimal reconstruction of dynamical systems: A noise amplification approach. Physical Review E 84, 016223](https://doi.org/10.1103/PhysRevE.84.016223).
+"""
+function estimate_maximum_delay(s::Vector{T}; tw=1:50, samplesize::Real=1) where {T<:Real}
+    L = zeros(T,length(tw))
+    # loop over time windows
+    cnt = 1
+    for i in tw
+        Y_act = embed(s,i,1)
+        L[cnt] = uzal_cost(Y_act; samplesize = samplesize)
+        cnt +=1
+    end
+    ~, τ_max = findmin(L)
```

Comment:
What's happening here? Why do you use the function `~` in such a way?

Reply:
I actually just want to supress the output, since I don't need it for any computations. That is still Matlab-syntax I suppose. Is there a similar, clean way in Julia?

Reply:
Yes, use `_` instead.